packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -76,20 +77,46 @@ export interface BuildFlags {
   message?: string;
 }
 
-const platformToGraphQLResourceClassMapping: Record<
-  Platform,
-  Record<UserInputResourceClass, BuildResourceClass>
+const iosUserInputResourceClassToBuildResourceClassMapping: Record<
+  UserInputResourceClass,
+  BuildResourceClass
 > = {
-  [Platform.ANDROID]: {
-    [UserInputResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
-    [UserInputResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
-  },
-  [Platform.IOS]: {
-    [UserInputResourceClass.DEFAULT]: BuildResourceClass.IosDefault,
-    [UserInputResourceClass.LARGE]: BuildResourceClass.IosLarge,
-  },
+  [UserInputResourceClass.DEFAULT]: BuildResourceClass.IosDefault,
+  [UserInputResourceClass.LARGE]: BuildResourceClass.IosLarge,
+  [UserInputResourceClass.M1_EXPERIMENTAL]: BuildResourceClass.IosM1Large,
 };
 
+const androidUserInputResourceClassToBuildResourceClassMapping: Record<
+  Exclude<UserInputResourceClass, UserInputResourceClass.M1_EXPERIMENTAL>,
+  BuildResourceClass
+> = {
+  [UserInputResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
+  [UserInputResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+};
+
+function resolveResourceClass(
+  platform: Platform,
+  resourceClassInput: UserInputResourceClass
+): BuildResourceClass {
+  if (platform !== Platform.IOS && resourceClassInput === UserInputResourceClass.M1_EXPERIMENTAL) {
+    Errors.error(
+      `Resource class ${UserInputResourceClass.M1_EXPERIMENTAL} is only available for iOS builds`,
```

Comment:
Is this throwing an exception or does it exist here immediately? If it just exits here I don't think this is appropriate behavior, it would be fine it top-level command code, but not in util function that potentially can be used in other places. If that is the case you should throw an exception instead.

Reply:
It just exits here